packages/rocketchat-api/server/v1/groups.js

```diff
@@ -348,12 +348,27 @@ RocketChat.API.v1.addRoute('groups.info', { authRequired: true }, {
 
 RocketChat.API.v1.addRoute('groups.invite', { authRequired: true }, {
 	post() {
-		const findResult = findPrivateGroupByIdOrName({ params: this.requestParams(), userId: this.userId });
+		let findResult;
+		const canAddUserToAnyPrivateGroup = RocketChat.authz.hasPermission(this.userId, 'add-user-to-any-p-room');
+		const params = this.requestParams();
+		if (canAddUserToAnyPrivateGroup) {
+			if ((params.roomId && params.roomId.trim()) || (params.roomName && params.roomName.trim())) {
+				const idOrName = params.roomId || params.roomName;
+				findResult = RocketChat.models.Rooms.findOneByIdOrName(idOrName);
+			} else {
+				throw new Meteor.Error('error-room-param-not-provided', 'The parameter "roomId" or "roomName" is required');
+			}
+			if (!findResult || findResult.t !== 'p') {
+				throw new Meteor.Error('error-room-not-found', 'The required "roomId" or "roomName" param provided does not match any group');
+			}
+		} else {
+			findResult = findPrivateGroupByIdOrName({ params, userId: this.userId });
+		}
 
 		const user = this.getUserFromParams();
 
 		Meteor.runAsUser(this.userId, () => {
-			Meteor.call('addUserToRoom', { rid: findResult.rid, username: user.username });
+			Meteor.call('addUserToRoom', { rid: findResult._id, username: user.username });
```

Comment:
This gets the id of the object, which is valid when it's a room, but when the user doesn't have the 'add-user-to-any-p-room', the findResult object is still going to be a subscription.

It's better to assign the id to a variable inside the conditional (room._id when the user has the permission, or subscription.rid when the user doesn't have it) and then use the variable here.